### PR TITLE
feat: add integration tests for det-deploy local

### PR DESCRIPTION
**Pull request by davidhershey**
## Description
Added integration tests that test `det-deploy local`, and add them to the `test-e2e` workflow in CircleCI

## Test Plan

See CircleCI execute the new tests.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
Question: how do people feel about putting the tests in `deploy/tests` instead of `e2e_tests/tests`?  My thinking is that the code overlap is actually pretty small (`run_basic_experiment`) but conceptually they're pretty different test suites.

I could be convinced that the helper code we need from e2e_tests is really a blueprint for the sort of python SDK features that we need, and that we can leave these new tests in e2e_tests until those features are part of the product.

**Comment by davidhershey** [comment]
Yeah this decision was purely on overlapped / useful code sharing.  I considered copy pasting, but having the tools seemed useful.

Definitely down to move once we have an SDK available.  Or can move now and copy / paste if people feel strongly.

**Review by davetroiano — APPROVED** [review]
overall lgtm! it may be worth it to add some coverage for `master-up`, `master-down`, `agent-up`, `agent-down`

File: e2e_tests/tests/deploy/etc/master.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  save_experiment_best: 0
+  save_trial_best: 1
+  save_trial_latest: 1
+
+db:
+  user: postgres
+  host: determined-db
+  port: 5432
+  name: determined
+  # password: database_password
```
Comment (rb-determined-ai) [inline e2e_tests/tests/deploy/etc/master.yaml:14]: delete this line?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1374,6 +1427,17 @@ workflows:
...
+      - test-det-deploy:
+          name: test-det-deploy-local
+          requires:
+            - package-and-push-system-local
+          matrix:
+            parameters:
+              parallelism: [2]
+              mark: ["det_deploy_local"]
+              det-version: [$CIRCLE_SHA1]
+
```
Comment (rb-determined-ai) [inline .circleci/config.yml:1439]: Should this have a `filter: branches` on it so it only runs on the master branch?

**Review by rb-determined-ai — COMMENTED** [review]
Overall looks really good.

**Comment by rb-determined-ai** [comment]
I'd echo what @davetroiano said that it might be nice to test `master/agent`-`up/down`.

**Review by davetroiano — APPROVED** [review]

**Pull request merged**